keanu-project/src/main/java/io/improbable/keanu/util/io/DotSaver.java

```diff
@@ -199,7 +200,12 @@ private void setDotLabelWithValue(Vertex<? extends Tensor> vertex) {
                 try {
                     Vertex parentVertex = (Vertex) method.invoke(vertex);
                     GraphEdge parentEdge = new GraphEdge(vertex, parentVertex);
-                    edges.stream().filter(parentEdge::equals).findFirst().get().appendToLabel(parentName);
+                    Optional<GraphEdge> foundEdge = edges.stream().filter(parentEdge::equals).findFirst();
```

Comment:
Alternatively, 

```
foundEdge
   .orElseThrow(() -> new IllegalStateException("Did not find parent edge " + parentName))
   .appendToLabel(parentName);
```
Might not be as clear but it's more in the spirit of Optional.

Reply:
thanks, that's nice

Reply:
I've gone for this:

    GraphEdge foundEdge = edges.stream().filter(parentEdge::equals).findFirst()
        .orElseThrow(() -> new IllegalStateException("Did not find parent edge " + parentName));
    foundEdge.appendToLabel(parentName);